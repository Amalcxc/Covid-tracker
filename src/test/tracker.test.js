import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Tracker from '../components/tracker';
import store from '../redux/configureStore';

describe('test covid tracker component', () => {
  test(' it should have a image', () => {
    render(
      <Provider store={store}>
        <Router>
          <Tracker />
        </Router>
      </Provider>,
    );
    return expect(screen.getByTestId('img')).toBeInTheDocument;
  });
});

describe('test jest snapshot', () => {
  it('render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Tracker />
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});