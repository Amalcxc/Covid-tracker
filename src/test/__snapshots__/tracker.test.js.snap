// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test jest snapshot render correctly 1`] = `
<div>
  <div
    className="d-flex flex-column justify-content-center align-items-start img-class"
  >
    <h1
      className="header"
    >
      Covid Tracker
    </h1>
    <img
      alt="logo"
      className="logo col-4"
      data-testid="img"
      src="worldmap.png"
      style={
        Object {
          "width": "67%",
        }
      }
    />
  </div>
  <div
    className="row m-0 p-0"
  />
</div>
`;

exports[`test with jest snapshot render correctly 1`] = `
<div>
  <div
    className="d-flex flex-column justify-content-center align-items-start img-class"
  >
    <h1
      className="header"
    >
      Covid Tracker
    </h1>
    <img
      alt="logo"
      className="logo col-4"
      data-testid="img"
      src="worldmap.png"
      style={
        Object {
          "width": "67%",
        }
      }
    />
  </div>
  <div
    className="row m-0 p-0"
  />
</div>
`;
